### PR TITLE
[luci-interpreter] fix pal related cmakes

### PR DESCRIPTION
**Pull request by binarman**
This PR moves platform specific sources and headers to PAL
and adds warning message.

related issue: #7288

ONE-DCO-1.0-Signed-off-by: Alexander Efimov <a.efimov@samsung.com>

### CODE REVIEW

File: compiler/luci-interpreter/src/kernels/CMakeLists.txt [hunk 1]
```diff
@@ -18,20 +17,8 @@ include(${KERNEL_REGISTER_FILE})
...
-target_include_directories(${LUCI_INTERPRETER_KERNELS} SYSTEM PRIVATE
-        "${TensorFlowRuySource_DIR}"
-        "${TensorFlowGEMMLowpSource_DIR}"
-        "${TensorFlowEigenSource_DIR}"
-        "${TensorFlowSource_DIR}")
-
-if (NOT LUCI_MICRO)
-  target_link_libraries(${LUCI_INTERPRETER_KERNELS}
-          PUBLIC ${LUCI_INTERPRETER_CORE}
-          PRIVATE nncc_common Threads::Threads)
```
Comment (m-bronnikov) [inline compiler/luci-interpreter/src/kernels/CMakeLists.txt:30]: As I understand, linkage with Threads already done in *linux/pal.cmake*, but what happens with nncc_common?

File: compiler/luci-interpreter/src/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,9 @@
-include(${LUCI_INTERPRETER_PAL_DIR}/pal.cmake)
+include("${LUCI_INTERPRETER_PAL_DIR}/pal.cmake")
```
Comment (m-bronnikov) [inline compiler/luci-interpreter/src/CMakeLists.txt:1]: Q. This change is necessary?

Comment (binarman) [reply compiler/luci-interpreter/src/CMakeLists.txt:1]: nope, it is just-in-case change =)

Comment (binarman) [reply compiler/luci-interpreter/src/CMakeLists.txt:1]:
```suggestion
include(${LUCI_INTERPRETER_PAL_DIR}/pal.cmake)
```

File: compiler/luci-interpreter/src/kernels/CMakeLists.txt (hunk 1 above)
Comment (binarman) [reply compiler/luci-interpreter/src/kernels/CMakeLists.txt:30]: hmmmm, I missed this one.
This library does not provide any code, just some flags. I'll check if luci-micro builds with it.

Comment (binarman) [reply compiler/luci-interpreter/src/kernels/CMakeLists.txt:30]: looks like arm build works with nncc too!
bringing it back

Comment (m-bronnikov) [reply compiler/luci-interpreter/src/kernels/CMakeLists.txt:30]: Thanks) 
Please, note that I'm asking about `nncc_common` missing in  linux configuration. I think it's ok to reduce linkage size for mcu pal.

Comment (binarman) [reply compiler/luci-interpreter/src/kernels/CMakeLists.txt:30]: too late, I enabled it everywhere...
let's disable it if needed later =)

**Review by m-bronnikov — APPROVED** [review]
LGTM

**Review by SlavikMIPT — APPROVED** [review]
LGTM

**Pull request merged**